src/AMQPEventsSender.js

```diff
@@ -49,61 +56,49 @@ class AMQPEventsSender extends EventEmitter {
    * @returns {Promise}
    */
   async disconnect() {
-    if (this._channelPromise) {
-      let channel;
+    if (this._channel) {
```

Comment:
Same story with race conditions as above. Look into old code. It's a reason why it has been implemented with promises.